ergo-lib/src/wallet/signing.rs

```diff
@@ -170,6 +171,7 @@ pub fn sign_transaction(
     prover: &dyn Prover,
     tx_context: TransactionContext,
     state_context: &ErgoStateContext,
+    tx_hints: &TransactionHintsBag,
```

Comment:
We're missing a code path for signing a reduced transaction with hints. Let's add the same optional parameter as in `sign_transaction` above and in `Wallet::sign_transaction` and the second method in Wasm `Wallet::sign_reduced_transaction_multi`.